### PR TITLE
embed pages in a page

### PR DESCRIPTION
**Pull request by christianhugoch**
- added PageComponent, is visible in the pageeditor
- prepare the segment.contents in page run() with the user layout
- draw the page in renderLayout
- loosely copied from embedded views

### CODE REVIEW

File: packages/saltcorn-builder/src/components/storage.js [hunk 1]
```diff
@@ -147,7 +148,17 @@ export /**
             configuration={segment.configuration || {}}
           />
         );
-      } else if (segment.type === "action") {
+      } else if (segment.type === "page") {
```
Comment (glutamate) [inline packages/saltcorn-builder/src/components/storage.js:151]: you can get this for free if you declare Page.craft.related.fields and segment_type, see https://github.com/saltcorn/saltcorn/blob/master/packages/saltcorn-builder/src/components/elements/ViewLink.js#L262 - then there is no need for a case in the storage functions - but only if we don't need name

**Review by glutamate — CHANGES_REQUESTED** [review]
I think we don't need the name (i.e. the one that gets set to a random identifier), in that case the code can be simplified. As far as I remember it was only in the view that the name was needed for some specific reason.

File: packages/saltcorn-builder/src/components/elements/Page.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+export const PageSettings = () => {
+  const node = useNode((node) => ({
+    page: node.data.props.page,
+    name: node.data.props.name,
+    node_id: node.id,
+  }));
+  const {
+    actions: { setProp },
+    name,
```
[the remark this answers is not in the document]
Comment (christianhugoch) [reply packages/saltcorn-builder/src/components/elements/Page.js]: removed it

File: packages/saltcorn-builder/src/components/elements/Page.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        `Page: ${page}`
+      )}
+    </div>
+  );
+};
+
+export const PageSettings = () => {
+  const node = useNode((node) => ({
+    page: node.data.props.page,
+    name: node.data.props.name,
```
[the remark this answers is not in the document]
Comment (christianhugoch) [reply packages/saltcorn-builder/src/components/elements/Page.js]: removed it

File: packages/saltcorn-builder/src/components/storage.js (hunk 1 above)
Comment (christianhugoch) [reply packages/saltcorn-builder/src/components/storage.js]: added fields and type to related property

**Pull request merged**